compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/AwtEventFilter.desktop.kt

```diff
@@ -19,25 +19,42 @@ package androidx.compose.ui.awt
 import java.awt.event.KeyEvent
 import java.awt.event.MouseEvent
 
-internal interface AwtEventFilter {
-    fun shouldSendMouseEvent(event: MouseEvent): Boolean = true
-    fun shouldSendKeyEvent(event: KeyEvent): Boolean = true
+internal interface AwtEventListener {
```

Comment:
Maybe `AwtInputEventListener` instead.

Reply:
@igordmn 
But I'm not even sure "listener" is better than "filter", given that the work it actually does is filtering. At the place where it's declared and called – its purpose is a filter. If there's one implementation that uses it as a listener then it's better to either call that implementation a mild hack, or add another "listener" interface.

Note that the original "filter" for which I implemented this also has (and modifies) state.

Reply:
> given that the work it actually does is filtering

Now, it does "handling" and "consuming"

> either call that implementation a mild hack

We either shouldn't use a hack, or change the abstraction, as we done in this PR. Keeping the hack is bad for code quality.

> or add another "listener" interface.

Maybe, but I am not sure that is worth to rewrite, it looks just different to me, not better.

> Note that the original "filter" for which I implemented this also has (and modifies) state.

But it modifies it for the filtering purpose, which is a valid case. In the other case - we call `onOutsideCallback` which is a side effect, and isn't valid for `Filter` abstraction.

Reply:
> Now, it does "handling" and "consuming" (as in "standard" event handling approaches)

The consuming in the "standard" event handling approaches is because there's an inherent hierarchy through which the event propagates. In our case the order of "listeners" is only defined by the order we put them in. I'm afraid that mixing listening and filtering is going to cause bugs because some "listener" didn't expect a previous "filter" to consume the event.

P.S. Really, the "standard" event handling approach is also a hack; it's just a hack that has become standard.